tests/models/plants/conftest.py

```diff
@@ -90,35 +90,37 @@ def fixture_canopy_layer_data():
     TODO: This is currently convoluted because of the way in which layer_heights is set
     within the plants model.
     """
+    lyr_str = fixture_core_components.layer_structure
+
     return {
         "layer_heights_full": (
             "layer_heights",
             np.array([32, 30, 20, 10, 0.1, -0.5, -1]),
-            np.array([0, 1, 2, 3, 11, 12, 13]),
+            np.logical_or(lyr_str.index_filled_atmosphere, lyr_str.index_all_soil),
         ),
         "layer_heights_canopy": (
             "layer_heights",
             np.array([32, 30, 20, 10]),
-            np.array([0, 1, 2, 3]),
+            np.logical_or(lyr_str.index_above, lyr_str.index_filled_canopy),
         ),
         "leaf_area_index": (
             "leaf_area_index",
             np.array([1, 1, 1]),
-            np.array([1, 2, 3]),
+            lyr_str.index_filled_canopy,
         ),
         "layer_fapar": (
             "layer_fapar",
             np.array([0.4, 0.2, 0.1]),
-            np.array([1, 2, 3]),
+            lyr_str.index_filled_canopy,
         ),
         "layer_absorbed_irradiation": (
             "layer_absorbed_irradiation",
```

Comment:
Do you consider the absorption by the topsoil?

Reply:
It should do, will look.

Reply:
OK - as discussed, since this is specifically the fraction of PAR absorbed, it doesn't really make sense to assign a fAPAR value to the topsoil layer and we need to revisit the light penetration through the canopy to include SW downwelling radiation more widely.